### ISSUE TITLE
Grouping logic uses local timezone rather than the provided timezone

### ISSUE THREAD

**Issue by nothing-personal-gh** [amcharts/amcharts5#610]
**Bug description**

In the XY Chart with a DateAxis and a ValueAxis, the grouping logic with `groupIntervals` uses the local timezone (user's computer timezone) rather than the provided timezone with `root.timezone`. 

I am using `valueYGrouped: "open"` so the first value in each group should be used for the grouped period.

Steps to reproduce:

1. Set your computer's timezone to 'Indian Standard Timezone'.
2. Open https://codepen.io/trooper-isa/pen/VwxarLB?editors=1000
3. Notice the daily grouping and the start value 
Daily grouping (for Aug 27th): **125.87**

<img width="497" alt="image" src="https://user-images.githubusercontent.com/111722528/189470760-cf0cfa71-604f-4180-aade-d86fb474381c.png">

Raw values at Aug 27th 00:00: **125.87**

<img width="486" alt="image" src="https://user-images.githubusercontent.com/111722528/189470732-f12888c5-ae83-4f80-bb3e-235fc8df13f5.png">

This value `125.87` comes from this data point:

```
    {
      "value": 125.87,
      "time": 1661538600000
    }
```

1661538600000 is `Saturday, 27 August 2022 00:00:00 [GMT+05:30]`. This makes sense as this is the first data point for IST.

4. Uncomment line 70 (`root.timezone = am5.Timezone.new("Australia/Sydney");`) in the codepen file.

Notice the daily grouping and the first value is different now.

Daily grouping (for Aug 27th): **125.87**

<img width="488" alt="image" src="https://user-images.githubusercontent.com/111722528/189471072-6a479794-fd72-494b-9b01-66275eba3a21.png">

Raw values at Aug 27th 00:00: **986.13**

<img width="503" alt="image" src="https://user-images.githubusercontent.com/111722528/189471087-6f25fbc7-1406-4041-ae91-bef963863a2e.png">

This value `986.13` comes from this data point:

```
    {
      "value": 986.13,
      "time": 1661522400000
    }
```

1661522400000 is `Saturday, 27 August 2022 00:00:00 [GMT+10:00]`. 

**Bug 1 - Default grouping** - Expected daily grouping value for 27th August is `986.13` as this is the first datapoint in the Sydney timezone. But the actual value is `125.87`.

**Bug 2 - Custom grouping** - If you add `groupDataCallback` and inspect the `time` field of the first `dataItem.get('originals').dataContext`, it is 00:00 of IST instead of AEST.

 - amCharts version - 5
 - Browser and its version - Chrome latest
 - Related frameworks and their versions - N/A

**Comment by zeroin** [comment]
It was a complicated one, thanks for the heads up. We will release an update soon and will let you know.

**Comment by martynasma** [comment]
Fixed in [5.2.22](https://github.com/amcharts/amcharts5/releases/tag/5.2.22).

## [5.2.22] - 2022-09-15

### Added
- ZigZag indicator added to `StockChart`.

### Changed
- `"%"` number formatting modifier will now consider locale when choosing whether percent sign goes before or after the number. At this moment only `tr_TR` (Turkish) locale is placing percent sign before number.
- `StockChart` indicators, that were using black (`0x000000`) color as a default series color now use `alternativeBackground` color of a theme (they'll now be white if "Dark" theme is used).

### Fixed
- Stock toolbar dropdown items will now display proper `title` attribute.
- HTML tooltips were flashing an unpositioned content briefly.
- The `"Z"` date formatting option was not taking into account Root element's `timezone` setting.
- If a timezone was set on a chart and data grouping was on, the grouping was not paying attention to the timezone.
- If a timezone which was not using DST was set on a chart, and user was in a timezone which used DST, the axis labels were shifted by 1 hour in some setups.
- `GaplessDateAxis` was not working properly with a timezone set.
- `DateAxis` fills that should have been visible were hidden in some cases.
- `StockChart` was incorrectly switching to percent mode when last of the compared series was removed.
- `"p"` number formatting modifier added. Works the same way as `"%"` except it treats numbers as absolute values and does not multiply them by 100.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
